repos/fdbt-site/src/pages/api/periodValidity.ts

```diff
@@ -1,25 +1,33 @@
 import { NextApiResponse } from 'next';
-import { updateSessionAttribute } from '../../utils/sessions';
-import { PERIOD_EXPIRY_ATTRIBUTE } from '../../constants/attributes';
+import { getSessionAttribute, updateSessionAttribute } from '../../utils/sessions';
+import {
+    MATCHING_JSON_ATTRIBUTE,
+    MATCHING_JSON_META_DATA_ATTRIBUTE,
+    PERIOD_EXPIRY_ATTRIBUTE,
+} from '../../constants/attributes';
 import { redirectToError, redirectTo, getAndValidateNoc } from '../../utils/apiUtils';
 import { ErrorInfo, NextApiRequestWithSession } from '../../interfaces';
 import { getFareDayEnd } from '../../data/auroradb';
-import { PeriodExpiry } from '../../interfaces/matchingJsonTypes';
+import { PeriodExpiry, Ticket, WithIds } from '../../interfaces/matchingJsonTypes';
+import { putUserDataInProductsBucketWithFilePath } from '../../utils/apiUtils/userData';
 
 export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
     try {
         const errors: ErrorInfo[] = [];
         if (req.body.periodValid) {
             const { periodValid } = req.body;
-            let { productEndTime } = req.body;
+
+            let productEndTime = '';
```

Comment:
Why are we doing this? Doesn't this mean we aren't taking the value the user entered?

Reply:
yes, it is not getting input from the user and taking from the DB call at later point of code.